integration/record.sh

```diff
@@ -0,0 +1,336 @@
+  git_url='gitspam.spam/spam/spam'
+
+  echo "
+apiVersion: tekton.dev/v1
+kind: PipelineRun
+metadata:
+  name: ${name}
+spec:
+  params:
+  - name: IMAGE_DIGEST
+    value: ${image_digest}
+  - name: IMAGE_URL
+    value: ${IMAGE_URL}
+  - name: TEST_OUTPUT
+    value: missing
+  - name: commit
+    value: ${git_commit_id}
+  - name: committer-date
+    value: ${git_committer_date}
+  - name: url
+    value: ${git_url}
+"
+}
+
+function wait_for_pipelinerun() {
+  local pr_name
+  local value
+
+  pr_name="$1"
+
+  log "Waiting for PipelineRun ${pr_name} to finish"
+  tkn pipelinerun logs "${pr_name}"
+
+  log "Waiting for Chains to process the PipelineRun ${pr_name} ..."
+  while :
+  do
+      value="$(oc get pr "${pr_name}" -o yaml | yq '.metadata.annotations."chains.tekton.dev/signed"')"
+      if [[ "$value" == "true" ]]; then
+          break
+      fi
+      if [[ "$value" == "false" ]]; then
+          log "❌ Uh oh! Chains was not able to sign PipelineRun ${pr_name}"
+          exit 1
+      fi
+      sleep 1
+  done
+
+  log "✅ PipelineRun ${pr_name} completed and processed by Chains"
+}
+
+function download_attestation() {
+  local image_digest
+  image_digest="$1"
+
+  cosign download attestation "${IMAGE_URL}@${image_digest}" \
+    | jq '.payload | @base64d | fromjson'
+}
+
+function setup_scenario() {
+  local name
+  local output_dir
+
+  name="$1"
+  log "🏃 Executing scenario ${name}"
+
+  output_dir="integration/recordings/${name}"
+  mkdir -p "${output_dir}"
+
+  echo -n "${output_dir}"
+}
+
+function run_in_cluster_pipeline() {
+  local output_dir
+  local image_digest
+  local pr_name
+
+  output_dir="$1"
+
+  render_pipeline | tee "${output_dir}/pipeline.yaml" | oc apply -f -
+
+  image_digest="$(build_image)"
+
+  pr_name="simple-build-run-$(rand_string)"
+
+  render_pipelinerun "${pr_name}" "${image_digest}" | \
+    yq '.spec.pipelineRef |= {"name": "simple-build"}' | \
+    oc create -f -
+
+  wait_for_pipelinerun "${pr_name}"
+
+  oc get pipelinerun "${pr_name}" -o yaml > "${output_dir}/pipelinerun.yaml"
+
+  download_attestation "${image_digest}" > "${output_dir}/attestation.json"
+}
+
+function run_inline_pipeline() {
+  local output_dir
+  local pipeline_spec
+  local image_digest
+  local pr_name
+
+  output_dir="$1"
+
+  pipeline_spec="$(render_pipeline | yq '.spec' -o json | jq -c )"
+
+  image_digest="$(build_image)"
+
+  pr_name="simple-build-run-$(rand_string)"
+
+  render_pipelinerun "${pr_name}" "${image_digest}" \
+    | yq ".spec.pipelineSpec |= ${pipeline_spec}" \
+    | oc create -f -
+
+  wait_for_pipelinerun "${pr_name}"
+
+  oc get pipelinerun "${pr_name}" -o yaml > "${output_dir}/pipelinerun.yaml"
+
+  download_attestation "${image_digest}" > "${output_dir}/attestation.json"
+}
+
+#########
+# Setup #
+#########
+log "Using repository ${IMAGE_URL}"
+cd "$(git root)"
+# Just a quick check to make sure there is an active connection to the cluster.
+oc get serviceaccount > /dev/null
```

Comment:
`kubectl cluster-info` could work here as well, something like:
```suggestion
echo |kubectl cluster-info
```